bots/tests-invoke

```diff
@@ -83,11 +84,26 @@ class PullTask(object):
         self.sink = None
         self.github_status_data = None
 
-    def start_publishing(self, host):
+    def detect_collisions(self, opts):
         api = github.GitHub()
         if not self.github_revision:
             self.github_revision = self.revision
 
+        if opts.pull_number:
+            pull = api.get("pulls/{0}".format(opts.pull_number))
+            if pull and pull['head']['sha'] != self.revision:
+                return 'Newer revision available on GitHub for this pull request'
+
+        statuses = api.get('commits/' + self.github_revision + '/statuses')
```

Comment:
Another whimsical thing, but please use `.format()` for these more complicated strings, and (possibly) `+` for the simple concatenatino in line 93 :-)